MathlibTest/FieldSimp.lean

```diff
@@ -1,12 +1,509 @@
+
+/-- info: P (y * y) -/
+#guard_msgs in
+example : P (y * (y + x) ^ (0:ℤ) * y) := by test_field_simp
+
+/-- error: simp made no progress -/
+#guard_msgs in
+example : P (x / y) := by test_field_simp
+
+/-- error: simp made no progress -/
+#guard_msgs in
+example : P (x / -y) := by test_field_simp
+
+/-- error: simp made no progress -/
+#guard_msgs in
+example : P (-x / y) := by test_field_simp
+
+/-- info: P ((x + y * (x + 1) / (y + 1)) / (x + 1)) -/
+#guard_msgs in
+example (hx : x + 1 ≠ 0) : P (x / (x + 1) + y / (y + 1)) := by test_field_simp
+
+/-- info: P ((x * (y + 1) + y * (x + 1)) / ((x + 1) * (y + 1))) -/
+#guard_msgs in
+example (hx : 0 < x + 1) (hy : 0 < y + 1) : P (x / (x + 1) + y / (y + 1)) := by test_field_simp
+
+-- TODO (new implementation): exploit `AtomM` to combine like terms, i.e. `x * y * x` to `x ^ 2 * y`
+
+/-- info: P (x * x / y) -/
+#guard_msgs in
+example : P (x / (y / x)) := by test_field_simp
+
+/-- info: P (x * (y ^ 3 * x)) -/
+#guard_msgs in
+example : P (x / (y ^ (-3:ℤ) / x)) := by test_field_simp
+
+/-- info: P (x * y ^ 3 * x) -/
+#guard_msgs in
+example : P ((x / y ^ (-3:ℤ)) * x) := by test_field_simp
+
+/-- info: P (x * y * x ^ 2 * y ^ 3) -/
+#guard_msgs in
+example : P (x ^ 1 * y * x ^ 2 * y ^ 3) := by test_field_simp
+
+/-- info: P (x * y * x ^ 2 / y) -/
+#guard_msgs in
+example : P (x ^ 1 * y * x ^ 2 * y⁻¹) := by test_field_simp
+
+-- TODO (new implementation): this should reduce to `P (1 / y)`
+/-- error: simp made no progress -/
+#guard_msgs in
+example (hx : x ≠ 0) : P (x / (x * y)) := by test_field_simp
+
+-- TODO (new implementation): this should reduce to `P 1`
+/-- error: simp made no progress -/
+#guard_msgs in
+example (hx : x ≠ 0) (hy : y ≠ 0) : P ((x * y) / (y * x)) := by test_field_simp
+
+-- TODO (new implementation): this should reduce to `P 1`
+/-- info: P (x * y / (y * x)) -/
+#guard_msgs in
+example (hx : x ≠ 0) (hy : y ≠ 0) : P ((x * y) * (y * x)⁻¹) := by test_field_simp
+
+-- TODO (new implementation): this should reduce to `P (x ^ 3)`
+/-- info: P (x * y * x ^ 2 / y) -/
+#guard_msgs in
+example (hy : y ≠ 0) : P (x ^ 1 * y * x ^ 2 * y⁻¹) := by test_field_simp
+
+/- ### Three atoms -/
+
+/-- error: simp made no progress -/
+#guard_msgs in
+example : P (x * y * z) := by test_field_simp
+
+/-- error: simp made no progress -/
+#guard_msgs in
+example : P (x * y + x * z) := by test_field_simp
+
+-- TODO (new implementation): this should reduce to `P (1 / (y + z))`
+/-- error: simp made no progress -/
+#guard_msgs in
+example (hx : x ≠ 0) : P (x / (x * y + x * z))  := by test_field_simp
+
+-- TODO (new implementation): this should reduce to `P (x / (x * (y + z)))`
+/-- error: simp made no progress -/
+#guard_msgs in
+example : P (x / (x * y + x * z))  := by test_field_simp
+
+end
+
+/-! ## Cancel denominators from equalities -/
+
+/-! ### Most common use case: Cancel denominators to something solvable by `ring`
+
+When (eventually) this is robust enough, there should be a `field` tactic
```

Comment:
Could you add an explanation here that `TODO (new implementation)` refers to this not-yet-extant `field` tactic?

Reply:
`TODO (new implementation)` just signifies tasks that should be in scope for `field_simp`, but that would require switching `field_simp` to a different implementation (e.g. our WIP re-implementation).

It's not referring to the idea I mention here, that there should be a `field` tactic.